AdobeMediaGallery/Plugin/Wysiwyg/Images/Storage.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeMediaGallery\Plugin\Wysiwyg\Images;
+
+use Magento\AdobeMediaGalleryApi\Api\Data\AssetInterface;
+use Magento\AdobeMediaGalleryApi\Model\Asset\Command\GetListByPathInterface;
+use Magento\AdobeMediaGalleryApi\Model\Asset\Command\DeleteByIdInterface;
+use Magento\Cms\Model\Wysiwyg\Images\Storage as StorageSubject;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\Directory\ReadInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * Ensures that metadata is removed from the database when a file is deleted and it is an image
+ */
+class Storage
+{
+    /**
+     * @var GetListByPathInterface
+     */
+    private $getMediaListByPath;
+
+    /**
+     * @var DeleteByIdInterface
+     */
+    private $deleteMediaAssetById;
+
+    /**
+     * @var Filesystem
+     */
+    protected $filesystem;
```

Comment:
Magento Health Index test results helped to identify this unnecessary protected scope
```suggestion
    private $filesystem;
```